Test/Core/Utility/GarbageCollectionUtils.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using NUnit.Framework;
+
+namespace Duality.Tests.Utility
+{
+	public class GarbageCollectionUtils
+	{
+		// No inlining is important here to defeat any optimizations the compiler can make that could cause instances to live longer.
+		[MethodImpl(MethodImplOptions.NoInlining)]
+		public static void CheckIfCleanedUp(Func<CleanupTestSet> func)
+		{
+			CleanupTestSet cleanupTestSet = func();
+
+			// In some cases unloading something happens async and can take some time such as with AssemblyLoadContext. Workaround this by retrying a couple of times..
+			for (int i = 0; cleanupTestSet.WeakReferences.Any(x => x.IsAlive) && i < 10; i++)
+			{
+				GC.Collect(2, GCCollectionMode.Forced, true);
+				GC.WaitForPendingFinalizers();
+			}
+			foreach (var item in cleanupTestSet.WeakReferences)
+			{
+				Assert.False(item.IsAlive);
+			}
+		}
+	}
+
+	public class CleanupTestSet
```

Comment:
Consider making this a private nested class, since it's an implementation detail.

Reply:
Its used in VertexBatchTest so that won't compile

Reply:
Ah, makes sense. In that case, please move it to a separate file which is named after the class. I always struggle to find stuff and get an overview in projects where multiple types are mashed in the same file, and I doubt I'm the only one there.

I'd also suggest to (but this is on you, would approve either way) change the class name to something that suggests its specific field of use and relation to GC - maybe something like `GarbageCollectionTestSet`?

Reply:
Yup I will

Also I agree thats a much better name :)